akka-http-core/src/main/scala/akka/http/ServerSettings.scala

```diff
@@ -46,7 +46,10 @@ final case class ServerSettings(
 
 object ServerSettings extends SettingsCompanion[ServerSettings]("akka.http.server") {
   final case class Timeouts(idleTimeout: Duration,
+                            requestTimeout: Duration,
                             bindTimeout: FiniteDuration) {
+    require(idleTimeout > Duration.Zero, "idleTimeout must be infinite or > 0")
+    require(requestTimeout > Duration.Zero, "requestTimeout must be infinite or > 0")
     require(bindTimeout > Duration.Zero, "bindTimeout must be > 0")
   }
```

Comment:
note to self: In some modules we moved away from case classes for "config types" as adding a field is binary-hard.
Perhaps this is something we should consider before calling http-core stable in 2.4.2?